pylint/extensions/magic_number.py

```diff
@@ -0,0 +1,82 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Checks for magic numbers instead of literals."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker, utils
+from pylint.checkers.base.comparison_checker import ComparisonChecker
+from pylint.interfaces import HIGH
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+LEGAL_MAGIC_NUMBERS = {-1, 0, 1, "", "__main__"}
+
+
+class MagicNumberChecker(BaseChecker):
+    """Checks for constants in comparisons."""
+
+    name = "magic-number"
+    msgs = {
+        "R2004": (
+            "Unnamed constant %s is used, consider using literal instead of magic numbers.",
+            "magic-comparison",
```

Comment:
```suggestion
            "magic-value-used-in-comparison",
```
I think the accepted software engineering term for unnamed constant is "magic values" or "magic numbers"

Reply:
I went with `magic-number-comparison`